lib/coverage.js

```diff
@@ -551,7 +546,32 @@ internals.file = async function (filename, data, options) {
     };
 
     // Use sourcemap consumer rather than SourceMapSupport.mapSourcePosition itself which perform path transformations
-    const sourcemap = options.sourcemaps ? SourceMapSupport.retrieveSourceMap(ret.filename) : null;
+    let sourcemap = null;
+    if (options.sourcemaps) {
+        sourcemap = SourceMapSupport.retrieveSourceMap(ret.filename);
+        if (!sourcemap) {
+            const smre = /\/\/\#.*data:application\/json[^,]+base64,.*\r?\n?$/;
+            let sourceIndex = data.source.length - 1;
+            while (sourceIndex >= 0 && !smre.test(data.source[sourceIndex])) {
+                sourceIndex--;
+            }
+
+            if (sourceIndex >= 0) {
```

Comment:
Can you add a comment to explain more of the purpose of line 550-570

Reply:
The line I removed, `const transformedFile = content.replace(/\/\/\#(.*)\r?\n?$/, '');`, used to remove inlined sourcemaps from the data.source object. `SourceMapSupport.retrieveSourceMap(ret.filename)` will only find inlined sourcemaps if they are present on disk. In combination with the Transform API, inlined sourcemaps are never found.
This code will look at the data.source object to find the inlined sourcemap from the specified Transform if the source disk file has no sourceMapURL.

Reply:
The test I added covers this case by:
1) Ensuring the source file has no sourceMapURL on disk
2) Injecting a sourceMapURL with inline source map data via the Transform API
3) Verifying that the original line numbers are present and accurate in the coverage analysis

Reply:
Thank you for the explanation. I'll merge and verify that I don't encounter any breakage. I'm considering this a bug fix unless you think otherwise. Thanks again!